cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramRuntimeProvider.java

```diff
@@ -47,7 +47,7 @@
    * The execution mode of the program runtime system.
    */
   enum Mode {
-    LOCAL, DISTRIBUTED
+    IN_MEMORY, LOCAL, DISTRIBUTED
```

Comment:
The program runner provider is not for this purpose, as what you described is implementation specifics. The ProgramRunnerProvider is intended to be generic so that can be implemented for different program runtime. Whether it needs to be wrapped or not for the state publishing purpose is internally to CDAP.